src/accapi/client.py

```diff
@@ -276,15 +266,27 @@ def request_hud_page(self, pageName: str):
         )
 
     def _run(self):
-        while not self._stopSignal:
-            try:
-                messageType = self._receive("B")
-            except ConnectionResetError:
-                self._update_connection_state("lost")
-                break
-            self._receiveMethods[messageType]()
-        self._socket.close()
-        self._socket = None
+        self._request_connection()
+        try:
+            while not self._stopSignal:
+                try:
+                    message = self._socket.recv(self._udpBufferSize)
```

Comment:
Ideally, the client would not be responsible for managing the receive buffer. In #4, a special class has been created that manages this buffer, leaving the client to do exactly what it used to do.